### DISCUSSION TITLE
Plotting decision analysis graph for different age groups

### DISCUSSION THREAD

**Discussion by mvb17** [MSKCC-Epi-Bio/decisioncurveanalysis#9] [Q&A]
Hello i would like to plot a decision analysis graph to predict 10 year probability  for different age groups 

I can use - stpm2- however, I quite like the probability curve you produce.

Procedure type is a binary variable 0 or 1
 Is it possible to produce different 10 year probabilities for different age groups is <50, 50-60, 60-70, 70+  for procedure 0 and procedure 1?

Pls see code below, I haven't inserted mean values for the different age groups - as this is the variable that probably needs tweaking? 
Appreciate any advice.

code: 
stcox procedure_type gender age institution 

*insert mean values 
gen modelgender = gender
gen modelinsitution = institution 

stcox procedure_type modelgender modelinstution, basesurv(surv_function) 

sum modelgender 
replace modelgender=r(mean)

*Linear predictor for risk 
predict xb, xb
sum xb, detail
predict see, stdp 

*baselien survival
sum surv_func if t<=10 

*convert to probability
gen risk10y=1-`base'^exp(xb) //converts to probability 
lab var risk10y "Probability of failure at 10 years" 

*Plot 10 year probability 
twoway (line risk10y procedure_type, sort clpat(solid solid) clwidth (medthick thin) clcolor (gs0 gs0)), title (10 year probability(%), margin(medium)) ylabel(75(5)100, angle(horizontal) nogrid) xtitle (years, margin(medium)) (150)250) legend(off) scheme (s2mono) grpahregion(color(white))

**Comment by VickersA** [comment]
I'm not sure what this has to do with decision curve analysis. It seems like you just want a graph of probability for different age groups.
